inc/Engine/Common/Clock/WPRClock.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace WP_Rocket\Engine\Common\Clock;
+
+class WPRClock implements ClockInterface {
+
+	/**
+	 * Retrieves the current time based on specified type.
+	 *
+	 *  - The 'mysql' type will return the time in the format for MySQL DATETIME field.
+	 *  - The 'timestamp' or 'U' types will return the current timestamp or a sum of timestamp
+	 *    and timezone offset, depending on `$gmt`.
+	 *  - Other strings will be interpreted as PHP date formats (e.g. 'Y-m-d').
+	 *
+	 * If `$gmt` is a truthy value then both types will use GMT time, otherwise the
+	 * output is adjusted with the GMT offset for the site.
+	 *
+	 * @since 1.0.0
+	 * @since 5.3.0 Now returns an integer if `$type` is 'U'. Previously a string was returned.
+	 *
+	 * @param string   $type Type of time to retrieve. Accepts 'mysql', 'timestamp', 'U',
+	 *                       or PHP date format string (e.g. 'Y-m-d').
+	 * @param int|bool $gmt  Optional. Whether to use GMT timezone. Default false.
+	 *
+	 * @return int|string Integer if `$type` is 'timestamp' or 'U', string otherwise.
+	 */
+	public function current_time( string $type, $gmt = 0 ) {
+		$current_time = current_time( $type, $gmt );
+		if ( ( is_string( $current_time ) && strtotime( $current_time ) ) || is_int( $current_time ) ) {
+			return apply_filters( 'rocket_current_time', $current_time );
+		}
+		return apply_filters( 'rocket_current_time', 0 );
```

Comment:
If I understand this part well, here we are checking if it's string (it can be string if gmt argument is not true) and it's a real date/datetime string by checking the return of strtotime (this returns false in case it's not valid value)
OR
WP Core function returns integer in case of gmt is true.
THEN
current_time is valid and apply the filter on this value.
OTHERWISE
apply the filter on the default value of zero.

But in our code we always use gmt=true so all what we do here is to make it more generic, correct?

I have three comments here:
1. This is a critical part which affects basic functionality of RUCSS and sending jobs so I believe we don't need filter here.
2. We need to define the filter only one time
3. it'd be easier to get the $gmt argument into the condition and typecast the filter return (if we really need the filter).

Reply:
Safeguard of filters have been changed. I misunderstood how it works :)